src/diffusers/models/attention_processor.py

```diff
@@ -2274,8 +2274,7 @@ def __call__(
                 query, key, value, attn_mask=attention_mask, dropout_p=0.0, is_causal=False
             )
 
-        hidden_states = hidden_states.transpose(1, 2).reshape(batch_size, -1, attn.heads * head_dim)
-        hidden_states = hidden_states.to(query.dtype)
+        hidden_states = hidden_states.transpose(1, 2).reshape(batch_size, -1, attn.heads * head_dim).to(query.dtype)
```

Comment:
Why do we need this change? For improvements to the library, feel free to open a separate PR :)